ush/find_last_cycle.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+
+#------------------------------------------------------------------------
+# find_last_cycle.sh
+#
+#   Test the common options for monitor data storage ($tankdir/stats),
+#   then determine the last cycle time for which there is data.  The
+#   last cycle time will be echoed to stdout. 
+#
+#   Extracted DA monitor data files are stored in a directory identified 
+#   as $tankdir, however there are 3 common directory structures in use.
+#   This script handles all 3.
+#
+#   The directory structures are:
+#      Operations:  $tankdir/$net/version/$run.$pdy/$hh/atmos/$monitor
+#      Developers*:  $tankdir/$monitor/stats/$net/$run.$pdy/$hh
+#      Monitors**:  $tankdir/stats/$net/$run.$pdy/$hh/$monitor
+#
+#   The get_stats_path.sh script will be used to find a valid path from 
+#   $tankdir to the directory above $run.$pdy (which, conveniently, is
+#   common to all cases). 
+#
+#
+#   * using global_workflow and rocoto
+#   ** This used to be the operations format, now used internally in 
+#      the DA monitors via the copy scripts. 
+#------------------------------------------------------------------------
+
+
+#---------------------------------------------------------
+#  Set default values and process command line arguments.
+#
+net=""
+run=gdas
+tankdir=""
+monitor=""
+
+while [[ $# -ge 1 ]]
+do
+   key="$1"
+
+   case $key in
+      -m|--mon)
+         monitor="$2"
+         shift # past argument
+      ;;
+      -n|--net)
+         net="$2"
+         shift # past argument
+      ;;
+      -r|--run)
+         run="$2"
+         shift # past argument
+      ;;
+      -t|--tank)
+         tankdir="$2"
+         shift # past argument
+      ;;
+   esac
+
+   shift
+done
+
+
+#-----------------------------------------
+# Call get_stats_path to extend $tankdir
+#
+path=`${MON_USH}/get_stats_path.sh --net $net --run $run --tank $tankdir --mon $monitor --rpath base`
+
+if [[ ${#path} -gt 0 ]]; then
+
+   hrs="18 12 06 00"
+  
+   #------------------------------------------------------------- 
+   #  Get list of subdirs which is $run.$pdy in all cases.  Then
+   #  step through the possible options for ops, dev, and mon
+   #  halting when the first valid directory structure is found.
+   #
+   flist=`find "${path}" -maxdepth 1 -mindepth 1 -type d -name "${run}.*" -printf "%f\n" 2>/dev/null`
+
+   lcyc=""
+   for file in $flist; do
+
+      for hr in $hrs; do
+
+         if [[ -d ${path}/${file}/${hr}/atmos/${monitor} ||     #ops
```

Comment:
The global workflow writes the ozone stats to ${path}/$file}, e.g. monitor/oznmon/stats/<experiment name>/gdas.20220202.  This should be added to the search path here.

Reply:
Agreed.  I'll add that case.

Reply:
@DavidHuber-NOAA  Updates have been made, and the workflow directory structure has been tested.  Please take another look when you have a chance.  Thanks.

Reply:
The default rocoto workflow location has changed as a result of https://github.com/NOAA-EMC/global-workflow/pull/1065.  The default location is now monitor/oznmon/stats/gdas.20220202/cyc, or what I had previously referred to as "dev" in  ush/get_stats_path.sh.  I've changed that to "wkf" and removed the old workflow case.

Reply:
Noted.  I will run a test case on S4 with the updated locations and give this another whirl.

Reply:
It's still running.  By 'run another test case', I meant I'm running the updated global workflow for 5 cycles.  I will run the monitor after that.  It should finish up today.

Reply:
Very good.  Sorry, didn't mean to press.

Reply:
OK, the test finished and everything looks good.  I think you can mark this resolved now.  Thanks!